catalog/ui/catalog-ui-search/src/test/java/org/codice/ddf/catalog/ui/security/accesscontrol/TagAggregationVisitorTest.java

```diff
@@ -648,4 +648,46 @@ public void testNestedAndOrFilterWithNoTagsAndOnlyTagsElsewhereNegatedEntirely()
     filter.accept(tagAggregationVisitor, null);
     assertThat(tagAggregationVisitor.getTags(), is(empty()));
   }
+
+  /*
+   * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * *
+   * Logical Redundancies
+   * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * *
+   */
+
+  @Test
+  public void testTagsPropagateThroughAndAndRedundancy() {
```

Comment:
✏️ 
```suggestion
  public void testTagsPropagateThroughDoubleAndRedundancy() {
```
or something similar. When I initially read it, I was confused. I don't have an overtly strong opinion on this though. Same comment for the tests below